schema.ts

```diff
@@ -1135,6 +1134,103 @@ export interface IndexDefinition {
   popularityRankProperty?: PropertyIdentifier<string>;
 }
 
+/**
+ * Determines how permissions are handled for this object.
+ */
+export enum PermissionsBehavior {
+  /**
+   * The object will inherit permissions from its parent.
+   */
+  Inherit = 'Inherit',
+}
+
+export interface BaseParentDefinition {
+  /**
+   * The identity of the parent object.
+   */
+  parentIdentity: IdentityDefinition;
+
+  /**
+   * Determines how permissions are handled for this object.
+   */
+  permissions?: PermissionsBehavior;
+}
+
+/**
+ * Defines that instances of this object will not be generated by crawling parents. Instead, parents for this
+ * will be specified in the {@link NotCrawlableParentDefinition.parentIdProperty}.
+ */
+export interface NotCrawlableParentDefinition extends BaseParentDefinition {
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that
+   * identifies the parent of this object.
+   *
+   * ValueType.Array is not supported is not supported for parentIdProperty.
+   *
+   * If this property references a property with {@link ValueType.Object} and
+   * {@link ValueHintType.Reference}, then it refers to an object in another logical table.
+   *
+   * Otherwise, it refers to an object in the same table.
+   */
+  parentIdProperty: PropertyIdentifier<string>;
+}
+
+export interface ParentChildParameterMapping {
+  /**
+   * The name of the parameter for the child sync table.
+   */
+  childParameterName: string;
+
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that
+   * identifies the parent of this object.
+   */
+  parentProperty: PropertyIdentifier<string>;
+}
+
+/**
+ * Determines how the parent object should be crawled for children.
+ */
+export enum CrawlingBehavior {
+  /**
+   * The parent object should be crawled for children.
+   */
+  Enabled = 'Enabled',
+}
+
+export enum LifecycleBehavior {
+  /**
+   * The child objects should be deleted when the parent object is deleted.
+   */
+  Inherit = 'Inherit',
+}
+
+/**
+ * Defines that this object has a parent object that will be crawled for children.
+ */
+export interface CrawlableParentDefinition extends BaseParentDefinition {
+  /**
+   * Determines how the parent object should be crawled for children.
+   */
+  crawling: CrawlingBehavior;
+
+  /**
+   * Determines how the lifecycle of the child objects should be handled.
+   */
+  lifecycle: LifecycleBehavior;
```

Comment:
I thought one of the motivations of subitems is that `lifecycle` would apply there too, so that parent deletes could be cascaded?

Reply:
I wasn't planning on building that support out of the gate, so didn't want to put it in the SDK.